src/models/matrix.rs

```diff
@@ -103,13 +103,15 @@ impl Model for Matrix {
     // / ]);
     // / ```
     // /
-    fn slide(&mut self, direction: Directions) {
+    fn slide(&mut self, direction: Directions) -> bool {
+        let old_board = self.board.clone();

```

Comment:
wouldn't toggle boolean inside the helpers be more efficient and increase performance?